onnxscript/tests/function_libs/torch_lib/ops_test_data.py

```diff
@@ -734,6 +735,11 @@ def _where_input_wrangler(
 
 
 SKIP_XFAIL_SUBTESTS: tuple[ops_test_common.DecorateMeta, ...] = (
+    skip(
+        "abs_complex",
+        matcher=lambda sample: sample.input.dtype == torch.float32,
```

Comment:
I wonder how general should we make on dtype in ops_test.py right now given that we only test float32.

Reply:
Potentially leverage the dtypes field (https://github.com/titaiwangms/onnx-script/blob/1f6d993ba8a19f5213c015e99a3081a12dfb1777/onnxscript/tests/function_libs/torch_lib/ops_test_common.py#L86), and create a constant of something like `REAL_TYPES`. In skip we can use `skip(..., dtypes=REAL_TYPES)`

`REAL_TYPES = (*BOOL_TYPES, *INT_TYPES, *FLOAT_TYPES)`

Alternatively, we can create a new mapping (TESTED_COMPLEX_OPS) and parameterize the test only for complex types. We can duplicate this test

https://github.com/titaiwangms/onnx-script/blob/1f6d993ba8a19f5213c015e99a3081a12dfb1777/onnxscript/tests/function_libs/torch_lib/ops_test.py#L322-L330

like so

```python
    @ops_test_common.add_decorate_info(
        ops_test_data.OPS_DB,
        "TestOutputConsistencyFullGraph",
        "test_output_match_opinfo_complex_types_",  # Update here
        skip_or_xfails=ops_test_data.EXPECTED_SKIPS_OR_FAILS,
    )
    @common_device_type.ops(  # type: ignore[misc]
        [info for info in ops_test_data.OPS_DB if info.name in ops_test_data.TESTED_COMPLEX_OPS],  # Update here
        allowed_dtypes=COMPLEX_TYPES,  # Update here
    )
    def test_output_match_opinfo_complex_types_(  # Update here
        self, device: str, dtype: torch.dtype, op: opinfo_core.OpInfo
    ):
        """Base test method for testing each op by running the full ONNX graph."""
        run_test_output_match(self, device, dtype, op, ops_test_common.graph_executor)
```

This way we may not need as many xfails, and we don't need to duplicate opinfos.

Same thing can be done for the test in `TestOutputConsistencyEager`

Reply:
Very nice!